### ISSUE TITLE
What's the syntax for minifying URLs?

### ISSUE THREAD

**Issue by JoeUX** [kangax/html-minifier#667]
Hi all – Thanks for the great work. The one thing I can't figure out is how to set the options for URL minification, which is using relateurl. I looked at the relateurl repo, but I'm struggling to understand the explanations offered there.

This option isn't covered in the blog post that goes through each option. Can you post or reply with the basics?

Thanks,

Joe

**Comment by alexlamsl** [comment]
As stated in [documentation](https://github.com/kangax/html-minifier#options-quick-reference), we use [relateurl](https://github.com/stevenvachon/relateurl) for that functionality. So refer to their documentation on how to use it 😉

**Comment by JoeUX** [comment]
As I said I already did that. I couldn't make heads or tails of that documentation. It's not explicit about defaults and options, and doesn't give examples of each option.

**Comment by alexlamsl** [comment]
Hmmm... would our [test cases](https://github.com/kangax/html-minifier/blob/gh-pages/tests/minifier.js#L1913-L1951) help you?

**Comment by JoeUX** [comment]
Possibly. I don't yet understand all of it, but the first block seems to be doing root-relative and then a path-relative scheme. I'll need to spend more time with it to understand how the test cases work (I'm new to JS).

What's the syntax for the keys and values of the relateurl options? My entrypoint is either the [HTMLMinifier website](http://kangax.github.io/html-minifier/) or the [Minify plugin](https://github.com/tssajo/Minify) for Sublime Text. I checked out the command line help doc, and it says `--minify-urls [value]`.

Do all values start with the string `RelateUrl` like in `RelateUrl.ROOT_RELATIVE`? That's how all the output options are portrayed on the [relateurl site](https://github.com/stevenvachon/relateurl), but none of the other (non-output) options seem to have that syntax. Some of the options have values themselves, like rejectedSchemes, so is this going to be a table/hash structure? (I'm not sure what an "object" is in the context of HTMLMinifier or JS in general – the HTMLMinifier doc gives object as a possible type for relateurl's value.)

**Comment by alexlamsl** [comment]
The web and the command-line interfaces can take the URL directly from the input text box and transform into a simple options object. You can also provide the full options on the command-line via stringified JSON or as part of [`--config-file`](https://github.com/kangax/html-minifier/blob/gh-pages/cli.js#L165).

I don't know if we have any extra modes of operations other than shortening URLs that would still resolve to the same resources given the minified document's location.

**Comment by Kristinita** [comment]
@JoeUX , `type:question`  

Have you solved your problem? Did you make a working example?

Thanks.

**Comment by JoeUX** [comment]
Hi @Kristinita, it's been a while but I don't think I solved my problem. I'll look into it again.

**Comment by JoeUX** [comment]
Also @Kristinita, I gave up on getting relateurl to work because I decided that I wanted full URLs in my webpage source. This was because I want my articles and blog posts to be standalone HTML documents – I wanted people to be able to download my articles and posts as HTML or MHTML files that they can read offline whenever they want.

This is probably a rare behavior, but I download pages as MHTML all the time and I wanted others to be able to do it (there's a flag in Chrome, Opera, and Vivaldi to enable MHTML downloads). For my pages to work properly this way, they need full URLs, since a browser that's opening a local HTML/MHTML file won't know the original page's URL. Having partial URLs would cause images, fonts, and other assets to not load.
